### PR TITLE
Check json format firstly

### PR DESCRIPTION
**Pull request by feihujiang**
#7943

### CODE REVIEW

**Comment by feihujiang** [comment]
@lavalamp  Sorry, I opened this new PR and closed #7944. Could you please review this one?

File: pkg/conversion/meta.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ type SimpleMetaFactory struct {
 // Interpret will return the APIVersion and Kind of the JSON wire-format
 // encoding of an object, or an error.
 func (SimpleMetaFactory) Interpret(data []byte) (version, kind string, err error) {
+	var f interface{}
+	err = json.Unmarshal(data, &f)
```
Comment (lavalamp) [inline pkg/conversion/meta.go]: As I said in #7944, "This isn't the right place to do this. Dig into the DataVersionAndKind method call below; it already parses w/ json package. This is a hot code path and we can't add an entirely new unmarshal call."

File: pkg/conversion/meta.go [hunk 2]
```diff
@@ -59,7 +59,13 @@ func (SimpleMetaFactory) Interpret(data []byte) (version, kind string, err error
 	}{}
 	err = json.Unmarshal(data, &findKind)
 	if err != nil {
-		return "", "", fmt.Errorf("couldn't get version/kind: %v", err)
+		return "", "", fmt.Errorf("unable to parse json: %v", err)
```
Comment (lavalamp) [inline pkg/conversion/meta.go]: This is the right place, but please don't add the below two checks (notice the tests fail now-- those aren't actually error conditions).

I'd recommend making this error message read thusly:
"couldn't get version/kind; json parse error: %v"

**Comment by feihujiang** [comment]
@lavalamp OK, I changed according to your advice.  Could you please review this PR again?

**Comment by lavalamp** [comment]
LGTM, thanks for the change!

**Pull request merged**